openassessment/management/tests/test_create_oa_submissions_from_file.py

```diff
@@ -0,0 +1,357 @@
+        assert_create_test_users_call(mock_call_command.call_args_list[0], {USERNAME_1, USERNAME_2}, False)
+        assert_create_test_users_call(mock_call_command.call_args_list[1], {STAFF_USER_1, STAFF_USER_2}, True)
+
+    def test_submit_ora_test_data(self):
+        """ Test for behavior of the submit step """
+        self.cmd.submit_ora_test_data(COURSE_ID, CONFIG_1)
+
+        # User 1 should have a submission, their workflow should be 'done', they should have a staff grade
+        # and they should be locked.
+        user_1_submission = sub_api.get_submissions(student_item(USERNAME_1, self.mock_block.location))[0]
+        user_1_workflow = workflow_api.get_workflow_for_submission(user_1_submission['uuid'], None)
+        assert user_1_workflow['status'] == 'done'
+        user_1_assessment = staff_api.get_latest_staff_assessment(user_1_submission['uuid'])
+        assert user_1_assessment['points_earned'] == 1
+        assert user_1_assessment['scorer_id'] == anonymous_user_id(STAFF_USER_2)
+        assert user_1_assessment['feedback'] == SUBMISSION_CONFIG_1['gradeData']['overallFeedback']
+        user_1_lock_owner = SubmissionGradingLock.get_submission_lock(user_1_submission['uuid']).owner_id
+        assert user_1_lock_owner == anonymous_user_id(STAFF_USER_1)
+
+        # User 2 should have a submission, their workflow should be 'waiting', they should not have a
+        # staff grade and they should not be locked
+        user_2_submission = sub_api.get_submissions(student_item(USERNAME_2, self.mock_block.location))[0]
+        user_2_workflow = workflow_api.get_workflow_for_submission(user_2_submission['uuid'], None)
+        assert user_2_workflow['status'] == 'waiting'
+        user_2_assessment = staff_api.get_latest_staff_assessment(user_2_submission['uuid'])
+        assert user_2_assessment is None
+        assert SubmissionGradingLock.get_submission_lock(user_2_submission['uuid']) is None
+
+    def test_reset_ora_test_data(self):
+        """ Test for behavior of the reset step"""
+        self.cmd.reset_ora_test_data(COURSE_ID, CONFIG_1)
+        assert self.mock_block.clear_student_state.call_count == 2
+
+        for reset_user in [USERNAME_1, USERNAME_2]:
+            self.mock_block.clear_student_state.assert_any_call(
+                anonymous_user_id(reset_user),
+                COURSE_ID,
+                str(self.mock_block.location),
+                anonymous_user_id(SUPERUSER_USERNAME)
+            )
+
+
+class CreateSubmissionsFromFileCallCommandTest(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        for username in [USERNAME_1, USERNAME_2, STAFF_USER_1, STAFF_USER_2, SUPERUSER_USERNAME]:
+            UserFactory.create(username=username)
+
+        self.mock_block = Mock(**MOCK_BLOCK_KWARGS)
+
+        def mock_anonymous_id_for_user(user, course_id):  # pylint: disable=unused-argument
+            return anonymous_user_id(user.username)
+
+        read_config_patcher = patch.object(
```

Comment:
Did this here rather than inline to avoid having to do four long daisy chained context managers